Auction Create Page Develop

FioreFlower opened the pull request:
## PR 타입(하나 이상의 PR 타입을 선택해주세요)
- [x] 기능 추가
- [x] 의존성, 환경 변수, 빌드 관련 코드 업데이트

<br/>

## 반영 브랜치
jinyeong/23-create-an-auction-registration-page -> main

<br/>

## 👨🔧 변경사항
<!-- 이 PR에서 어떤점들이 변경되었는지 기술해주세요. 가급적이면 as-is, to-be를 활용해서 작성해주세요.  -->
- **AS-IS**
  - 기존 디자인은 일반적인 인풋 요소를 사용해서 이미지를 받았습니다.
  - 기존의 인풋 요소로는 사용자에게 충분한 시각 정보와 편의성을 제공할 수 없음
  - 사용자 입장에서 0번째 인덱스 파일이 메인 이미지가 되는 것을 인지하고 조작하기 어려움
  - 파일을 나눠서 업로드 할 수 없고 한번에 5개 파일을 업로드 해야만 해서 불편함
  - 사용자가 이미지 순서를 변경하기 어려움
- **TO-BE**
  - 문제 해결을 위해서는 첫 이미지가 메인 이미지가 된다는 정보를 간결하게 사용자에게 알려야함
  - 파일을 나눠서 업로드하기 편리해야함
  - 이미지 인덱스를 사용자가 편리하게 변경할 수 있어야함
<br />

## 🎨 이미지 첨부
<!-- 이미지 첨부는 자유 입니다. 하실분은 하시고 안하실 분은 안하셔도 됩니다. -->
![화면 기록 2024-07-08 오전 1 00 47](https://github.com/Fan-Tion/Front-End/assets/78160955/b774f055-419a-49da-8982-9d87d2855447)

<br/>

## 👨💻 테스트
<!-- 본 변경사항이 테스트가 되었는지 기술해주세요 --> 
- [ ] 테스트 코드
- [ ] API 테스트
<br/>

## ➕ 이슈 링크

- [Front-End #23](https://github.com/Fan-Tion/Front-End/issues/23)

FioreFlower committed:
feat : 텍스트 에디터 추가, 드래그 가능한 이미지 업로드 UI 생성

- react-dnd, react-dropzone, react-dnd-html5-backend 라이브러리 추가
- 이미지 최대 5개까지 업로드, 드래그앤드롭으로 인덱스를 변경할 수 있음
- 메인 이미지가 될 이미지를 강조 표시함

#23 UI 부분 생성

FioreFlower committed:
fix : 업로드 가능한 파일 설정 추가

- 파일 입력시 이미지 파일만 허용하도록 변경
- 5MB로 업로드 가능한 용량 제한

#23 후속 수정

Min-byeonghyun reviewed (approved):
react-dnd와 react-dropzone을 사용한 이미지 업로드 인터페이스가 인상적입니다.
좋은 것 같습니다.

FioreFlower committed:
chore : Quill 대신 TOAST UI 사용하도록 변경

- Quill 관련 패키지 삭제

#23

FioreFlower committed:
feat : TOAST UI EDITOR 추가

- TOAST UI 패키지 설치
- editor 초기화
- 이미지 업로드를 못하게 하기 위해 툴바 커스텀
- 마크다운 사용할 수 없도록 커스텀 진행

# follow up #23

FioreFlower committed:
feat : TOAST UI 툴바 한글로 표시하도록 변경

# F/U #23

DoHyeongKwonn reviewed (approved):
잘 봤습니다 고생하셨습니다.

FioreFlower committed:
style : 경매 생성 페이지 view 생성

- 스타일 지정 작업 진행 중

F/U #23
